addContractURI

Guillaume-web opened the pull request:
Test commit

Guillaume-web committed:
addContractURI

marc-aurele-besner commented on test/SimpleNft.test.js:
```diff
@@ -126,9 +126,15 @@ describe('Simple NFT', function () {
+    await contract.connect(user1).setContractURI('Test New ContractURI');
+
+    const contractURI_final = await contract.contractURI();
+    console.log('contractURI_final', contractURI_final);
```
1er PR super 🥳 
Par contre, je te demanderais d'ajouté une assertion dans ton test pour confirmer que le changement fait en appelant setContractURI a bien été fait. Inspire-toi de la ligne 126 plus haut 😉

Guillaume-web committed:
add contractURI

marc-aurele-besner commented on test/SimpleNft.test.js:
```diff
@@ -126,9 +126,15 @@ describe('Simple NFT', function () {
-  // it('Does anyone can update contract uri? (should not)', async function () {
-  //   // Add test logic here
-  // });
+  it.only('Does anyone can update contract uri? (should not)', async function () {
```
Je te demanderais d'ôter le .only et pour les prochains test, crée un différent fichier de tests

Guillaume-web committed:
add contractURI finale

Guillaume-web committed:
Create guillaume.SimpleNft.t.js

marc-aurele-besner commented on test/guillaume.SimpleNft.t.js:
```diff
@@ -0,0 +1,137 @@
+    const contractURI_final = await contract.contractURI();
+
+    expect(await contractURI_final).to.equal('Test New ContractURI');
+  });
```
Il manque 
```
});
```
à la fin du fichier pour fermer le describe du début

Guillaume-web committed:
ajout de multiples fonctions

ajout de :

- Does contract owner can update base uri? (should be)
- Does anyone can update base uri? (should not)
- Does contract owner can update contract uri and base uri? (should be)
- Does anyone can update contract uri and base uri? (should not)
